java/ql/src/experimental/Security/CWE/CWE-691/InsecureClassLoading.qhelp

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+  <overview>
+    <p>
+      Android Contexts that depend on external inputs are potentially vulnerable to code execution attacks.          
+      This query detects the scenarios where an attacker controlled input is used to create a new Android <code> Context</code>. 
+    </p>
+  </overview>
+
+  <recommendation>
+    This can be avoided in two ways. 
+    <ol>
+      <li> 
+        To avoid arbitrary code execution, the application should sanitize all external inputs.
+      </li>
+      <li> Changing the application architecture to prevent unintended access to sensitive components.</li>
+
+
+  </recommendation>
+
+  <example>
+    <p>The following example shows a 'bad' way of creating Android `Context` variables. An attacker may register a package with the name `somename.plugin.` having a `pluginLoader`  class.</p>
+    <sample src="InsecureClassLoading.java" />
+    <p> Then, when the code above executes, the attacker would be able to run malicious code in the context of the applciaiton.</p>
+  </example>
+
+  <references>
+    <li>
+      <a href="https://cwe.mitre.org/data/definitions/749.html">CWE-749</a>
```

Comment:
Is there any Google security advice about plugin architectures you can link here?

Reply:
Unfortunately, I can find anything to suit my needs here so I am just ignoring it for now.